helm/hpcc/templates/NOTES.txt

```diff
@@ -53,41 +53,35 @@ This chart has defined the following HPCC components:
 
 {{- range $dali := (.Values.dali | default dict) }}
  {{- if and (not $dali.disabled) (hasKey $dali "services") -}}
-  {{- $sashaServices := $dali.services | default dict -}}
-  {{- if not $sashaServices.disabled -}}
-   {{- range $serviceName, $_service := $sashaServices }}
-    {{- $service := ($_service | default dict) -}}
-    {{- if not $service.disabled -}}
-     {{- if hasKey $names $serviceName -}}
-      {{- required ( printf "Component name %s in section sasha was already seen in section %s" $serviceName (get $names $serviceName)) nil -}}
-     {{- else }}
-  dali.sasha.{{ $serviceName }}
-      {{- $_ := set $names $serviceName (printf "dali.%s" $dali.name) -}}
-      {{- $localImage := include "hpcc.imageName" (dict "root" $ "me" $dali ) | quote }}
-      {{- if not ( eq $defaultImage $localImage ) -}}
-       {{- printf " using overridden image %s" $localImage -}}
-      {{- end }}
-     {{- end -}}
-    {{- end -}}
-   {{- end -}}
-  {{- end -}}
- {{- end -}}
-{{- end -}}
-
-{{- $sashaServices := $.Values.sasha | default dict -}}
-{{- if not $sashaServices.disabled -}}
- {{- range $serviceName, $_service := $sashaServices }}
-  {{- $service := ($_service | default dict) -}}
-  {{- if not $service.disabled -}}
+  {{- $daliSashaServicesCtx := dict "services" $dali.services | default dict -}}
+  {{- include "hpcc.getSashaServices" $daliSashaServicesCtx -}}
+  {{- range $serviceName, $_service := $daliSashaServicesCtx.services }}
+   {{- $service := ($_service | default dict) -}}
    {{- if hasKey $names $serviceName -}}
     {{- required ( printf "Component name %s in section sasha was already seen in section %s" $serviceName (get $names $serviceName)) nil -}}
    {{- else }}
-  sasha.{{ $serviceName }}
-    {{- $localImage := include "hpcc.imageName" (dict "root" $ "me" $service ) | quote }}
+  dali.sasha.{{ $serviceName }}
+    {{- $_ := set $names $serviceName (printf "dali.%s" $dali.name) -}}
+    {{- $localImage := include "hpcc.imageName" (dict "root" $ "me" $dali ) | quote }}
     {{- if not ( eq $defaultImage $localImage ) -}}
      {{- printf " using overridden image %s" $localImage -}}
     {{- end }}
    {{- end -}}
   {{- end -}}
- {{- end }}
+ {{- end -}}
+{{- end -}}
+
+{{- $sashaServicesCtx := dict "services" $.Values.sasha | default dict -}}
```

Comment:
These two bits of code are very similar, could they not be made common with some options?

Reply:
Again extra parenthesis showing what default applies to could improve readability.

Reply:
> These two bits of code are very similar, could they not be made common with some options?

they are but subtly different. I'll leave it as is for, it's a separate issue/can be done another time.

Reply:
> Again extra parenthesis showing what default applies to could improve readability.

agreed, will change.